aether-odk-module/aether/odk/api/xform_utils.py

```diff
@@ -974,19 +982,28 @@ def __get_all_paths(dictionary):
 
     It's only used to get the jsonpaths (or xpaths)
     of the instance skeleton defined in the xForm.
-
-    Assumption: there are no lists in the skeleton.
     '''
     def walk(obj, parent_keys=[]):
         for k, v in obj.items():
             is_dict = isinstance(v, dict)
+            is_list = isinstance(v, list)
             if k.startswith('@'):  # ignore attributes
                 continue
             keys = parent_keys + [k]
             xpath = '/' + '/'.join(keys)
-            paths.append((xpath, isinstance(v, dict)))
+            paths.append((xpath, is_dict or is_list))
             if is_dict:
                 walk(v, keys)
+            elif is_list:
+                # in pyxform 1.x.x there could be duplicated entries like:
```

Comment:
What if there aren't duplicates?

Reply:
If there is only one is the first condition, if they are different... we have a problem.

The reason of having it duplicated is this one: https://github.com/XLSForm/pyxform/pull/380